x-pack/plugins/observability_solution/synthetics/server/plugin.ts

```diff
@@ -77,6 +79,7 @@ export class Plugin implements PluginType {
       logger: this.logger,
       telemetry: this.telemetryEventsSender,
       isDev: this.initContext.env.mode.dev,
+      isServerless: this._isServerless,
```

Comment:
Q: Do you need to define the getter (`_isServerless()`) where you can use the initial context directly like in the `isDev` so replace it with: `this.initContext.env.packageInfo.buildFlavor === 'serverless'`

Reply:
Sure we could do that. In this case, it's just a matter of preference I carried over from the client plugin's implementation of this same functionality. The difference is, on the public plugin we are referencing this value several times and it made no sense to use the uglier reference code repeatedly.

Since we're only referencing the value one time here, it won't violate DRY principles if we set this value inline, instead of using the getter. I just find the encapsulation of that logic cleaner from a readability perspective. If someone isn't directly interacting with the `isServerless` piece of this class, the code is nicer to read in this way.

Reply:
I agree with Justin here. Even though the _isServerless is only used here by now, we might in the future add more code that needs this differentiation. And I see us copying the ugly check all over the place.

Reply:
Ok, I see. We use it only here so I wasn't sure if we need a separate function for that but it makes sense to have it if we plan to reuse it in the future.